src/main/webapp/resources/js/pages/projects/samples/components/SamplesMenu.jsx

```diff
@@ -0,0 +1,361 @@
+import React, { lazy, Suspense } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Button, Dropdown, Menu, message, Row, Space } from "antd";
+import {
+  addToCart,
+  clearFilterByFile,
+  downloadSamples,
+  exportSamplesToFile,
+  filterByFile,
+  reloadTable,
+} from "../../redux/samplesSlice";
+import { setBaseUrl } from "../../../../utilities/url-utilities";
+import {
+  validateSamplesForLinker,
+  validateSamplesForMerge,
+  validateSamplesForRemove,
+} from "../services/sample.utilities";
+import {
+  IconCloseSquare,
+  IconCloudDownload,
+  IconCloudUpload,
+  IconCode,
+  IconDropDown,
+  IconFile,
+  IconFileExcel,
+  IconPlusSquare,
+  IconShare,
+  IconShoppingCart,
+} from "../../../../components/icons/Icons";
+import {
+  CloseCircleOutlined,
+  FileTextOutlined,
+  MergeCellsOutlined,
+} from "@ant-design/icons";
+import { useGetProjectDetailsQuery } from "../../../../apis/projects/project";
+
+const MergeModal = lazy(() => import("./MergeModal"));
+const RemoveModal = lazy(() => import("./RemoveModal"));
+const CreateModal = lazy(() => import("./CreateNewSample"));
+const LinkerModal = lazy(() => import("./LinkerModal"));
+const FilterByFileModal = lazy(() => import("./FilterByFileModal"));
+
+/**
+ * React element to render a row of actions that can be performed on
+ * samples in the table
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export default function SamplesMenu() {
+  const dispatch = useDispatch();
+
+  const {
+    projectId,
+    selected,
+    selectedCount,
+    filterByFile: fileFiltered,
+  } = useSelector((state) => state.samples);
+  const { data: details = {} } = useGetProjectDetailsQuery(projectId);
+
+  const [mergeVisible, setMergeVisible] = React.useState(false);
+  const [removedVisible, setRemovedVisible] = React.useState(false);
+  const [createSampleVisible, setCreateSampleVisible] = React.useState(false);
+  const [linkerVisible, setLinkerVisible] = React.useState(false);
+  const [filterByFileVisible, setFilterByFileVisible] = React.useState(false);
+  const [sorted, setSorted] = React.useState({});
+
+  /**
+   * When a merge is completed, hide the modal and ask
+   * the table to reset
+   */
+  const onMergeComplete = () => {
+    setMergeVisible(false);
+    dispatch(reloadTable());
+  };
+
```

Comment:
Can you add comments for each of these constants

Reply:
Updated in c09c8da